examples/sync-local-to-remote/main.go

```diff
@@ -0,0 +1,327 @@
+	// clusters. Going with the example above if AppConfig is on the
+	// local cluster and AppInstance is on some provider clusters and
+	// AppDatabase on other provider clusters the manager needs all
+	// three (AppConfig, AppInstance, AppDatabase) in its schema while.
+	// The providers for the AppInstance and AppDatabase clusters only
+	// need their respective types.
+	managerScheme := runtime.NewScheme()
+	if err := corev1.AddToScheme(managerScheme); err != nil {
+		return fmt.Errorf("error adding corev1 to manager scheme: %w", err)
+	}
+
+	mgr, err := mcmanager.New(localCfg, provider, mcmanager.Options{
+		Scheme: managerScheme,
+	})
+	if err != nil {
+		return fmt.Errorf("error setting up mcmanager: %w", err)
+	}
+
+	// A label selector predicate to only sync ConfigMaps we create.
+	predicate, err := predicate.LabelSelectorPredicate(metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"example.io/propagate": "true",
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to build label selector predicate: %w", err)
+	}
+
+	// One controller that watches ConfigMaps in the local cluster and
+	// propagates them to all provider clusters.
+	if err := mcbuilder.ControllerManagedBy(mgr).
+		Named("local").
+		For(&corev1.ConfigMap{}, mcbuilder.WithEngageWithLocalCluster(true)).
+		WithEventFilter(predicate).
+		Complete(
+			pushLocalToProvider(mgr.GetLocalManager(), provider),
+		); err != nil {
+		return fmt.Errorf("failed to build controller: %w", err)
+	}
+
+	// Another controller that watches ConfigMaps in provider clusters
+	// and ensures they match the local cluster, reverting any local
+	// changes.
+	if err := mcbuilder.ControllerManagedBy(mgr).
+		Named("provider").
+		For(&corev1.ConfigMap{}, mcbuilder.WithEngageWithProviderClusters(true)).
+		WithEventFilter(predicate).
+		Complete(
+			pullProviderFromLocal(mgr.GetLocalManager(), provider),
+		); err != nil {
+		return fmt.Errorf("failed to build controller: %w", err)
+	}
+
+	return mgr.Start(ctx)
+}
+
+func pushLocalToProvider(localCl cluster.Cluster, provider *file.Provider) mcreconcile.Func {
+	return mcreconcile.Func(
+		func(ctx context.Context, req mcreconcile.Request) (ctrl.Result, error) {
+			log := ctrllog.FromContext(ctx).WithValues(
+				"cluster", req.ClusterName,
+				"namespace", req.Request.Namespace,
+				"name", req.Request.Name,
+			)
+			log.Info("Reconciling ConfigMap")
+
+			cm := &corev1.ConfigMap{}
+			if err := localCl.GetClient().Get(ctx, req.Request.NamespacedName, cm); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return reconcile.Result{}, err
+				}
+				log.Info("ConfigMap not found, deleting from provider clusters")
+				// Iterate over all provider clusters and delete
+				// the ConfigMap. Note that .ClusterNames is not
+				// part of the multicluster provider interface.
+				// Accessing this depends on the provider.
+				var errs error
+				for _, clusterName := range provider.ClusterNames() {
```

Comment:
wondering, this races (for dynamic providers). If a cluster comes online just in this moment, we will never sync to it. Do we have an easy way to trigger a resync to newly engaged clusters?

Not necessarily blocking this PR, but the example feels a little incomplete right now.